### PR TITLE
Drop support for typhoeus 0.4

### PR DESCRIPTION
**Pull request by aka47**
tyhoeus 0.4 was release almost 10 years ago and is long outdated. The current version is 1.4
Lets remove all patches and specs that target this version.

### CODE REVIEW

File: lib/vcr/library_hooks/typhoeus.rb [hunk 1]
```diff
@@ -2,137 +2,131 @@
...
-        # @private
-        class << self
-          def vcr_response_from(response)
-            VCR::Response.new \
-              VCR::ResponseStatus.new(response.code, response.status_message),
-              response.headers,
-              response.body,
-              response.http_version,
-              { "effective_url" => response.effective_url }
+        if ::Typhoeus::Request.method_defined?(:encoded_body)
```
Comment (olleolleolle) [inline lib/vcr/library_hooks/typhoeus.rb:67]: Q: Is this a part of the "new"/current API?

File: spec/monkey_patches.rb [hunk 2]
```diff
@@ -77,11 +71,11 @@ def disable_all!
   $original_typhoeus_global_hooks = Typhoeus.on_complete.dup
   $original_typhoeus_before_hooks = Typhoeus.before.dup
 elsif defined?(::Typhoeus::Hydra.global_hooks)
```
Comment (olleolleolle) [inline spec/monkey_patches.rb:73]: (Can probably drop the if conditional check here, too)

File: spec/monkey_patches.rb [hunk 3]
```diff
@@ -65,18 +65,10 @@ def disable_all!
   end
 end
 
-if defined?(::Typhoeus.before)
```
Comment (olleolleolle) [inline spec/monkey_patches.rb:74]: Ah, _this_ conditional is perhaps needed: we have some `Gemfile` variants where the Typheous gem is not defined, I believe.

**Comment by aka47** [comment]
I believe it looks good and can be merged..

**Review by olleolleolle — APPROVED** [review]
This drops patches and special support for an ancient version of Typheous.

**Comment by olleolleolle** [comment]
@aka47 Thanks for maintaining this!

**Pull request merged**